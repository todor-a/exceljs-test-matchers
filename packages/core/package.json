{
  "author": "Todor Andonov",
  "dependencies": {
    "table": "6.8.2"
  },
  "description": "",
  "devDependencies": {
    "vitest": "1.5.2"
  },
  "exports": {
    ".": {
      "import": {
        "default": "./build/index.js",
        "types": "./build/index.d.ts"
      },
      "require": {
        "default": "./build/index.cjs",
        "types": "./build/index.d.cts"
      }
    },
    "./package.json": "./package.json"
  },
  "files": [
    "build",
    "src"
  ],
  "keywords": [],
  "license": "ISC",
  "main": "build/index.cjs",
  "module": "build/index.js",
  "name": "@exceljs-test-matchers/core",
  "scripts": {
    "build": "pnpm run build:tsup",
    "build:tsup": "tsup",
    "clean": "rimraf ./build && rimraf ./coverage",
    "clean:all": "pnpm run clean && rimraf ./node_modules",
    "test:cspell": "cspell '**'",
    "test:eslint": "eslint --ext .ts,.tsx ./src",
    "test:types": "tsc",
    "test:unit": "vitest",
    "test:attw": "attw --pack"
  },
  "sideEffects": false,
  "type": "module",
  "types": "build/index.d.ts",
  "version": "1.0.0"
}
