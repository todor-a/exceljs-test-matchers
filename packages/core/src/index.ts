import * as ExcelJs from 'exceljs';

export const getSheetNames = (workbook: ExcelJs.Workbook) => {
    return workbook.worksheets.map(sheet => sheet.name);
};

export * from './findSheet';
export * from './hasSheetHeaders';
export * from './hasSheets';
export * from './isWorkbook';
export * from './toTable';
export * from './toWorkbook';
export * from './types';
