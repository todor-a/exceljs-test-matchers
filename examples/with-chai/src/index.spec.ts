import { toWorkbook } from '@exceljs-test-matchers/core';
import { expect } from 'chai';
import { describe, it } from 'mocha';

const sheet = 'sheet-a';

describe('toHaveSheets', () => {
    it('should be able to use the matcher', () => {
        const wb = toWorkbook([['a']], sheet);
        expect(wb).toHaveSheets([sheet]);
    });

    it('should return error when sheet missing', () => {
        const wb = toWorkbook([['a']], sheet);
        wb.addWorksheet('sheet-b');
        wb.addWorksheet('sheet-c');

        try {
            expect(wb).toHaveSheets(['sheet-a', 'sheet-w']);
        } catch (error) {
            expect(error).to.matchSnapshot();
        }
    });

    it(`should return error when sheets is present but shouldn't be`, () => {
        const wb = toWorkbook([['a']], sheet);

        try {
            expect(wb).not.toHaveSheets([sheet]);
        } catch (error) {
            expect(error).to.matchSnapshot();
        }
    });
});

describe('toHaveSheetHeaders', () => {
    it('should be able to use the matcher', () => {
        const wb = toWorkbook([['a']], sheet);
        const ws = wb.getWorksheet(sheet);

        expect(ws).toHaveSheetHeaders(['a']);
    });

    it('should return error when header not present', () => {
        const wb = toWorkbook([['a']], sheet);
        const ws = wb.getWorksheet(sheet);

        try {
            expect(ws).toHaveSheetHeaders(['q']);
        } catch (error) {
            expect(error).to.matchSnapshot();
        }
    });

    it('should return error when some headers are not present', () => {
        const wb = toWorkbook([['a']], sheet);
        const ws = wb.getWorksheet(sheet);

        try {
            expect(ws).toHaveSheetHeaders(['a', 'q']);
        } catch (error) {
            expect(error).to.matchSnapshot();
        }
    });

    it(`should return error when header is present but shouldn't be`, () => {
        const wb = toWorkbook([['a']], sheet);
        const ws = wb.getWorksheet(sheet);

        try {
            expect(ws).not.toHaveSheetHeaders(['a']);
        } catch (error) {
            expect(error).to.matchSnapshot();
        }
    });
});
