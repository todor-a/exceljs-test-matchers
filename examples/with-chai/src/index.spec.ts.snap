// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toHaveSheetHeaders should return error when header is present but shouldn't be 1`] = `[AssertionError: Expected the headers ["a"] not to be present, but all are found.]`;

exports[`toHaveSheetHeaders should return error when header not present 1`] = `[AssertionError: Expected the headers ["q"] to be present, but ["q"] are missing.]`;

exports[`toHaveSheetHeaders should return error when some headers are not present 1`] = `[AssertionError: Expected the headers ["a","q"] to be present, but ["q"] are missing.]`;

exports[`toHaveSheets should return error when sheet missing 1`] = `[AssertionError: Expected the sheets ["sheet-a","sheet-w"] to be present, but ["sheet-w"] are missing.]`;

exports[`toHaveSheets should return error when sheets is present but shouldn't be 1`] = `[AssertionError: Expected the sheets ["sheet-a"] not to be present, but they were found.]`;
